src/TumblThree/TumblThree.Applications/Downloader/AbstractDownloader.cs

```diff
@@ -310,6 +310,7 @@ protected virtual async Task<bool> DownloadBinaryPostAsync(TumblrPost downloadIt
                 UpdateProgressQueueInformation(Resources.ProgressDownloadImage, fileName);
                 if (!await DownloadBinaryFileAsync(fileLocation, fileLocationUrlList, Url(downloadItem)))
                 {
+                    RemoveFileFromDb(FileNameUrl(downloadItem)); // TODO: Temp plan, the addition should happen only when the download is successful.
```

Comment:
That could be a way to do it. But keep in mind that AddFileToDb is called early, because this url shall not be downloaded from another thread while this one is still working on it, and it also determines the final filename in case of using filename templates.

Reply:
At the moment, I have no better way to improve/staticize the `AddFileToDb` function. I'm not sure what you mean, but this PR is a proposal and not a final solution.

Reply:
I understood that it's a draft and work in progress (WIP). Everything is ok, I only wanted to give more information as input.